src/components/export.tsx

```diff
@@ -1,51 +1,214 @@
-import { useState } from 'react';
+import '@isrd-isi-edu/chaise/src/assets/scss/_export.scss';
+
+import { useEffect, useState } from 'react';
 import Dropdown from 'react-bootstrap/Dropdown';
 import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
 import Tooltip from 'react-bootstrap/Tooltip';
+
 import { MESSAGE_MAP } from '@isrd-isi-edu/chaise/src/utils/message-map';
+import { ConfigService } from '@isrd-isi-edu/chaise/src/services/config';
+import { LogService } from '@isrd-isi-edu/chaise/src/services/log';
+import { LogActions } from '@isrd-isi-edu/chaise/src/models/log';
+import ExportModal from '@isrd-isi-edu/chaise/src/components/export-modal';
+import useError from '@isrd-isi-edu/chaise/src/hooks/error';
+import { ChaiseAlertType } from '@isrd-isi-edu/chaise/src/providers/alerts';
+import useAlert from '@isrd-isi-edu/chaise/src/hooks/alerts';
 
 type ExportProps = {
-  reference: any,
-  disabled: boolean
-}
+  reference: any;
+  /**
+   * prop to make export button disable
+   */
+  disabled: boolean;
+};
 
+const Export = ({ reference, disabled }: ExportProps): JSX.Element => {
+  /**
+   * State variable to open progress modal dialog
+   */
+  const [showProgressModal, setShowProgressModal] = useState<boolean>(false);
+  /**
+   * State variable to export options
+   */
+  const [options, setOptions] = useState<Array>([]);
```

Comment:
Use `any[]` to be consistent with the `any` typeof `selectedOption`. Eventually when we have proper export object type in ermrestjs, we will fix these.